pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/DictionaryIndexConfig.java

```diff
@@ -21,29 +21,42 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
 import java.util.Objects;
 import javax.annotation.Nullable;
 import org.apache.pinot.spi.config.table.IndexConfig;
+import org.apache.pinot.spi.config.table.Intern;
 
 
 public class DictionaryIndexConfig extends IndexConfig {
 
-  public static final DictionaryIndexConfig DEFAULT = new DictionaryIndexConfig(false, false, false);
-  public static final DictionaryIndexConfig DISABLED = new DictionaryIndexConfig(true, false, false);
+  public static final DictionaryIndexConfig DEFAULT = new DictionaryIndexConfig(false, false, false, Intern.DISABLED);
+  public static final DictionaryIndexConfig DISABLED = new DictionaryIndexConfig(true, false, false, Intern.DISABLED);
 
   private final boolean _onHeap;
   private final boolean _useVarLengthDictionary;
+  private final Intern _intern;
 
-  public DictionaryIndexConfig(Boolean onHeap, @Nullable Boolean useVarLengthDictionary) {
-    this(false, onHeap, useVarLengthDictionary);
+  public DictionaryIndexConfig(Boolean onHeap, @Nullable Boolean useVarLengthDictionary, Intern intern) {
```

Comment:
In order to keep backward compatibility (and maybe reduce the number of changes), can you add a constructor with the older signature?

```
public DictionaryIndexConfig(Boolean onHeap, @Nullable Boolean useVarLengthDictionary) {
  this(onHeap, useVarLengthDictionary, null);
}
```